wolfcrypt/src/tfm.c

```diff
@@ -3834,7 +3834,7 @@ int fp_to_unsigned_bin_len(fp_int *a, unsigned char *b, int c)
   for (; x >= 0; x--) {
      b[x] = 0;
   }
-  if ((i < a->used - 1) || ((a->dp[i] >> j) != 0)) {
```

Comment:
When a->dp[i] >> j is not zero, and i < a->used, there was bytes not put into output array.
Need to check the second part only when i == a->used - 1